core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -351,13 +351,17 @@ class ControllerBrokerRequestBatch(controller: KafkaController, stateChangeLogge
     addUpdateMetadataRequestForBrokers(controllerContext.liveOrShuttingDownBrokerIds.toSeq, Set(topicPartition))
   }
 
-  def addStopReplicaRequestForBrokers(brokerIds: Seq[Int], topicPartition: TopicPartition, deletePartition: Boolean,
-                                      callback: (AbstractResponse, Int) => Unit) {
+  def addStopReplicaRequestForBrokers(brokerIds: Seq[Int], topicPartition: TopicPartition, deletePartition: Boolean): Unit = {
     brokerIds.filter(_ >= 0).foreach { brokerId =>
+      def callback(stopReplicaResponse: AbstractResponse): Unit = {
+        if (deletePartition && controllerContext.isTopicQueuedUpForDeletion(topicPartition.topic))
```

Comment:
When there is a pending partition reassignment task, a deleted topic will be queued up but not executed until the reassignment completes. So, we probably want to further check that isTopicIneligibleForDeletion(topic) is false.